scalyr_agent/scalyr_client.py

```diff
@@ -176,15 +176,28 @@ def __init__(
         }
 
         # Configure compression type
+        self.__compression_type = compression_type
+
         self.__compress = None
         encoding = None
 
+        if compression_level < 1 or compression_level > 9:
```

Comment:
Don't lz4 default to zero?  I think this would end up turning that to 9?

Reply:
Yep, good catch - I will change this and move it to a configuration layer where we validate those settings.

Reply:
Here we go - a09e62521f406da5ec6e1761d515eaf9a31f9b9b.

In ideal world, we would throw (I was never a fan of software silently accepting invalid values), but for backward compatibility reasons, I use a per compression algorithm specific default if an invalid value is specified in the config.